prboom2/src/p_map.c

```diff
@@ -2058,13 +2060,21 @@ dboolean PIT_RadiusAttack (mobj_t* thing)
   if (dist < 0)
   dist = 0;
 
-  if (dist >= bombdamage)
+  if (dist >= bombdistance)
     return true;  // out of range
 
   if ( P_CheckSight (thing, bombspot) )
     {
     // must be in direct path
-    P_DamageMobj (thing, bombspot, bombsource, bombdamage - dist);
+
+    // [XA] independent damage/distance calculation.
+    //      same formula as eternity; thanks Quas :P
+    if (bombdamage == bombdistance)
+      damage = bombdamage - dist;
+    else
+      damage = (bombdamage * (bombdistance - dist) / bombdistance) + 1;
```

Comment:
Can bombdistance be 0?

Reply:
Fortunately not -- the function always returns early if bombdistance is zero on the `//out of range` check a couple of lines before.